napari/utils/events/evented_model.py

```diff
@@ -94,6 +116,29 @@ def __new__(mcs, name, bases, namespace, **kwargs):
                 # required for pydantic>=1.8.0 due to:
                 # https://github.com/samuelcolvin/pydantic/pull/2064
                 EventedModel.__config__.json_encoders[f.type_] = encoder
+            # we now reinsert type hints for our own validations
+            # hijacking the key_field (cannot make new ones because of __slots__)
+            # which is otherwise unused
+            if f.type_ is EventedList:
```

Comment:
To be fair, this is not really something the *model* needs to know; it's just that we need `Field` to be initialized differently, and this initialization happens in the model. I guess it could be tackled with a field subclass as you mentioned above, but that seemed worse to me because you'd have to use that instead of plain pydantic fields... Thought maybe the `EventedModel` metaclass could take care of that conversion?